main.cpp

```diff
@@ -31,30 +31,25 @@ struct T
     std::string name;
 };
 
-struct CompareDaFunk                                //4
+struct CompareDaFunk                             //4
 {
-    T* compare(T* a, T* b) //5
+    const T* compare(const T& a, const T& b)
     {
-        if (a != nullptr && b != nullptr)
-        {
-            if( a->value < b->value ) return a;
-            if( a->value > b->value ) return b;
-        }
+        if( a.value < b.value ) return &a;
+        if( a.value > b.value ) return &b;
+
         return nullptr;
```

Comment:
The previous function's return type was `T*`
The instructions indicate that you should try to change pointers to references **wherever possible**.
They do not say to change the return type
- [ ] revise